pkg/model/iam/tests/iam_builder_master_strict.json

```diff
@@ -0,0 +1,165 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeInstances",
+        "ec2:DescribeRouteTables",
+        "ec2:DescribeSecurityGroups",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeVolumes"
+      ],
+      "Resource": [
+        "*"
+      ],
+      "Sid": "kopsK8sEC2MasterPermsDescribeResources"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:CreateRoute",
+        "ec2:CreateSecurityGroup",
+        "ec2:CreateTags",
+        "ec2:CreateVolume",
+        "ec2:DeleteSecurityGroup",
+        "ec2:DeleteVolume",
+        "ec2:ModifyInstanceAttribute"
+      ],
+      "Resource": [
+        "*"
```

Comment:
Can we limit?

Reply:
In some cases you can use the aws:RequestTags condition to force the creation of a resource (Volume/Security group) to contain a specific tag value (which you use in the next statement to ensure that you are only working on your own resources)
and then move the Delete* actions into the "with tagged resources" below.
with Create/ModifyTags you can use a ForAllValues:StringEquals condition on Aws::TagKeys to limit which tags this role allows you to play with (ie only the tags associated with Kubernetes)

Reply:
Some of the `Create*` calls will need to be updated to include request tags. A related Kubernetes issue has been raised here: https://github.com/kubernetes/kubernetes/issues/50898